engine/src/system_factory.cpp

```diff
@@ -0,0 +1,743 @@
+    static GFXColor planet_mat_specular  = vs_config->getColor( "default", "planet_mat_specular",
+                                           GFXColor( 0.0,0.0,0.0,1.0 ) );
+    static GFXColor planet_mat_emissive = vs_config->getColor( "default", "planet_mat_emmissive",
+                                           GFXColor( 0.0,0.0,0.0,1.0 ) );
+    GFXMaterial ourmat;
+    GFXGetMaterial( 0, ourmat );
+    setMaterialAmbient(ourmat, planet_mat_ambient);
+    setMaterialDiffuse(ourmat, planet_mat_diffuse);
+    setMaterialSpecular(ourmat, planet_mat_specular);
+    setMaterialEmissive(ourmat, planet_mat_emissive);
+
+    initializeMaterial(object, ourmat);
+    ourmat.power = getFloatAttribute(object, "SPower", ourmat.power);
+
+    // End planet not glow code
+
+    // "Invisible" object
+    if(object.attributes.count("reflectnolight"))
+    {
+        ourmat.sr = ourmat.sg = ourmat.sb = ourmat.dr = ourmat.dg =
+                ourmat.db = ourmat.ar = ourmat.ag = ourmat.ab = 0;
+    }
+
+    bootstrap_draw( "Loading "+fullname );
+
+    // Parse destinations (jump only?)
+    if(object.attributes.count("destination") )
+    {
+        destination = ParseDestinations( object.attributes["destination"] );
+        isDestination = true;
+    }
+
+    // Parse lights - for Suns
+    // Discussion - the original code supported multiple lights
+    // This is why we have curlights as a vector
+    // If you disable this, the ship in dock will be dark. Weird.
+    if(object.attributes.count("light") )
+    {
+        unsigned long index = 0;
+        char local = 0;
+        std::istringstream stream(object.attributes["light"]);
+        stream >> index >> local;
+
+        Light light = lights[index];
+        GFXLight gfx_light(true, Vector(0, 0, 0));
+        gfx_light.SetProperties(AMBIENT, light.ambient);
+        gfx_light.SetProperties(DIFFUSE, light.diffuse);
+        gfx_light.SetProperties(SPECULAR, light.specular);
+
+        GFXLightLocal local_light;
+        local_light.islocal = (local == 'l');
+        local_light.ligh = gfx_light;
+        curlights.push_back(local_light);
+    }
+
+    // Parse faction
+    if(object.attributes.count("faction") )
+    {
+        int originalowner =
+            FactionUtil::GetFactionIndex( UniverseUtil::GetGalaxyProperty( this->fullname, "faction" ) );
+        faction = FactionUtil::GetFactionIndex( object.attributes["faction"] );
+        if (faction == originalowner) {
+            int ownerfaction = FactionUtil::GetFactionIndex( UniverseUtil::GetGalaxyFaction( this->fullname ) );
+            faction = ownerfaction;
+        }
+    }
+
+    // Adjust speed and rotation
+    // Discussion - the original value (day/year) needs to be adjusted to velocity
+    // by multiplying
+    float float_pi = static_cast<float>(M_PI);
+    float float_year_scale = static_cast<float>(game_options.YearScale);
+    if(std::fabs(rotational_velocity) > .00001f)
```

Comment:
1. The `0.00001f` is nearing the limit of precision on a `float` type.
2. We shouldn't have magic numbers like that scattered about. At the very least, give them a name and make them a constant in the function; though we might want to promote them to constants in the class/file too as makes sense.

Reply:
not going to block on this; there's too much of it in the code. The refactor AS-IS is more important, just take the note above as  long term goal.

Also, we should document places where we might push the boundaries of the precision of a float, and watch out for any `float() == float()` or `double() == double()` comparisons as those will need to have a fuzzy comparison done `abs(A-B) < C` to be valid.

Reply:
I am unhappy with the whole float comparison thing. I did this to resolve a compiler warning or refactored existing code that did the same. I did a quick search just now and discovered no one has provided a function for this. Will add a TODO next to each, and will pick this up at a later time. Given the complexity of this subject (https://stackoverflow.com/questions/17333/what-is-the-most-effective-way-for-float-and-double-comparison for example), I'd rather not tackle this now.

Reply:
yeah, just making a note. The `>` and `<` operations are as big a deal, is the `==`, `>=`, and `<=` that are a problem.